Xamarin.Essentials/Calendars/Calendars.android.cs

```diff
@@ -405,6 +405,34 @@ static async Task<bool> PlatformUpdateCalendarEvent(CalendarEvent eventToUpdate)
             throw new ArgumentException("[Android]: Could not update appointment with supplied parameters");
         }
 
+        static async Task<bool> PlatformSetEventRecurrenceEndDate(string eventId, DateTimeOffset recurrenceEndDate)
+        {
+            await Permissions.RequestAsync<Permissions.CalendarWrite>();
+
+            var existingEvent = await GetEventByIdAsync(eventId);
+            CalendarEvent thisEvent;
+            if (existingEvent == null || string.IsNullOrEmpty(existingEvent.CalendarId))
+            {
+                return false;
+            }
+            else
```

Comment:
else is redundant because of the return statement in the if clause
Move the thisEvent declaration so you have
var thisEvent = await GetEventByIdAsync(eventId);

Reply:
Altered accordingly